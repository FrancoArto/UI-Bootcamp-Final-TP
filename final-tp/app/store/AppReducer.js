import { combineReducers } from 'redux';
import tweetsReducer from './tweets/tweetsReducer'
import trendsReducer from './trends/trendsReducer'
import settingsReducer from './settings/settingsReducer'
import userReducer  from './users/userReducer'

const AppReducer = combineReducers({
  tweetsReducer,
  trendsReducer,
  settingsReducer,
  userReducer  
});

export default AppReducer;