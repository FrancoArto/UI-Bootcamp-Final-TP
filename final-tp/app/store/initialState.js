const tweetsInitialState = {
    data: [],
    loading: false,
    loadingSingleTweet: false,
    error: null,
    searchText: null,
    searchResults: [],
    userTimeline: [],
    singleTweetId: null,
    singleTweet: {}
};

const userInitialState = {
    loading: false,
    error: null,
    userId: null,
    userData: {}
}

const trendsInitialState = {
    data: [],
    loading: false,
    error: null,
}

//False is for 'not checked'
const appViewsConfigs = {
    verified: false,
    following: false,
    defaultInfo: false,
    withLink: false,
    withTruncatedText: false
}


export { tweetsInitialState, appViewsConfigs, userInitialState, trendsInitialState };