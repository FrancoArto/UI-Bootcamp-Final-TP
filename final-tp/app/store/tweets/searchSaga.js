import { GET_SEARCHTWEETS_URL, GET_APPENDRESULTS_URL } from '../../api/apiUrls';
import { takeLatest, put, call, select, all } from 'redux-saga/effects';
import { FETCH_SEARCH_BEGIN, fetchSearchSuccess, fetchSearchError, FETCH_MORE_RESULTS_REQUEST, fetchMoreResultsSuccess, fetchMoreResultsFailure } from './searchTweetsActions';

const getSearchText = state => state.searchTweetsReducer.searchText
const getMaxId = state => state.searchTweetsReducer.data[state.searchTweetsReducer.data.length - 1].id
const count = 50


function* fetchSearchResults() {  
  try {
    const searchText = yield select(getSearchText)
    const response = yield call(fetch, GET_SEARCHTWEETS_URL(searchText))
    const data = yield call([response, "json"]);
    yield put(fetchSearchSuccess(data.statuses))
  } catch (er) {
    yield put(fetchSearchError(er))
  }
}

function* searchTweets() {
  yield takeLatest(FETCH_SEARCH_BEGIN, fetchSearchResults)   
}


function* fetchResultsToAppend() {
  try {
    const searchText = yield select(getSearchText)
    const maxId = yield select(getMaxId)
    const response = yield call(fetch, GET_APPENDRESULTS_URL(searchText, count, maxId))
    const results = yield call([response, "json"]);
    const resultArray = results.statuses
    const data = resultArray.slice(1)
    yield put(fetchMoreResultsSuccess(data))
  } catch (er) {
    yield put(fetchMoreResultsFailure(er))
  }
}

function* appendResults() {
  yield takeLatest(FETCH_MORE_RESULTS_REQUEST, fetchResultsToAppend)   
}



export function* searchSaga() {
  yield all([
    searchTweets(),
    appendResults()
  ])
}