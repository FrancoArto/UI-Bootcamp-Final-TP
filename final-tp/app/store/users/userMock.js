export default {
  "id": 8105922,
  "id_str": "8105922",
  "name": "Clarín",
  "screen_name": "clarincom",
  "location": "Buenos Aires, Argentina",
  "profile_location": null,
  "description": "Noticias de Argentina y el mundo las 24 horas. Información de último momento, opinión, análisis e informes especiales.",
  "url": "http://t.co/1RuczZo0FK",
  "entities": {
    "url": {
      "urls": [
        {
          "url": "http://t.co/1RuczZo0FK",
          "expanded_url": "http://www.clarin.com",
          "display_url": "clarin.com",
          "indices": [
            0,
            22
          ]
        }
      ]
    },
    "description": {
      "urls": []
    }
  },
  "protected": false,
  "followers_count": 2810171,
  "friends_count": 265,
  "listed_count": 10833,
  "created_at": "Fri Aug 10 15:14:11 +0000 2007",
  "favourites_count": 391,
  "utc_offset": null,
  "time_zone": null,
  "geo_enabled": true,
  "verified": true,
  "statuses_count": 301743,
  "lang": "es",
  "status": {
    "created_at": "Thu Dec 20 17:06:00 +0000 2018",
    "id": 1075799489686974500,
    "id_str": "1075799489686974467",
    "text": "Atacaron a balazos el Concejo Municipal de Rosario: \"Con la mafia no se jode\" https://t.co/B8NZ861uoF https://t.co/pliMCVpKd6",
    "truncated": false,
    "entities": {
      "hashtags": [],
      "symbols": [],
      "user_mentions": [],
      "urls": [
        {
          "url": "https://t.co/B8NZ861uoF",
          "expanded_url": "https://www.clarin.com/policiales/atacaron-balazos-concejo-deliberante-rosario-mafia-jode_0_t9aX8sX08.html?utm_term=Autofeed&utm_medium=Social&utm_source=Twitter#Echobox=1545324824",
          "display_url": "clarin.com/policiales/ata…",
          "indices": [
            78,
            101
          ]
        }
      ],
      "media": [
        {
          "id": 1075799486948098000,
          "id_str": "1075799486948098054",
          "indices": [
            102,
            125
          ],
          "media_url": "http://pbs.twimg.com/media/Du4B7v0WwAYZ4qc.jpg",
          "media_url_https": "https://pbs.twimg.com/media/Du4B7v0WwAYZ4qc.jpg",
          "url": "https://t.co/pliMCVpKd6",
          "display_url": "pic.twitter.com/pliMCVpKd6",
          "expanded_url": "https://twitter.com/clarincom/status/1075799489686974467/photo/1",
          "type": "photo",
          "sizes": {
            "thumb": {
              "w": 150,
              "h": 150,
              "resize": "crop"
            },
            "large": {
              "w": 1200,
              "h": 630,
              "resize": "fit"
            },
            "small": {
              "w": 680,
              "h": 357,
              "resize": "fit"
            },
            "medium": {
              "w": 1200,
              "h": 630,
              "resize": "fit"
            }
          }
        }
      ]
    },
    "extended_entities": {
      "media": [
        {
          "id": 1075799486948098000,
          "id_str": "1075799486948098054",
          "indices": [
            102,
            125
          ],
          "media_url": "http://pbs.twimg.com/media/Du4B7v0WwAYZ4qc.jpg",
          "media_url_https": "https://pbs.twimg.com/media/Du4B7v0WwAYZ4qc.jpg",
          "url": "https://t.co/pliMCVpKd6",
          "display_url": "pic.twitter.com/pliMCVpKd6",
          "expanded_url": "https://twitter.com/clarincom/status/1075799489686974467/photo/1",
          "type": "photo",
          "sizes": {
            "thumb": {
              "w": 150,
              "h": 150,
              "resize": "crop"
            },
            "large": {
              "w": 1200,
              "h": 630,
              "resize": "fit"
            },
            "small": {
              "w": 680,
              "h": 357,
              "resize": "fit"
            },
            "medium": {
              "w": 1200,
              "h": 630,
              "resize": "fit"
            }
          }
        }
      ]
    },
    "source": "<a href=\"https://www.echobox.com\" rel=\"nofollow\">Echobox Social</a>",
    "in_reply_to_status_id": null,
    "in_reply_to_status_id_str": null,
    "in_reply_to_user_id": null,
    "in_reply_to_user_id_str": null,
    "in_reply_to_screen_name": null,
    "geo": null,
    "coordinates": null,
    "place": null,
    "contributors": null,
    "is_quote_status": false,
    "retweet_count": 1,
    "favorite_count": 4,
    "favorited": false,
    "retweeted": false,
    "possibly_sensitive": false,
    "lang": "es"
  },
  "contributors_enabled": false,
  "is_translator": false,
  "is_translation_enabled": true,
  "profile_background_color": "131516",
  "profile_background_image_url": "http://abs.twimg.com/images/themes/theme14/bg.gif",
  "profile_background_image_url_https": "https://abs.twimg.com/images/themes/theme14/bg.gif",
  "profile_background_tile": true,
  "profile_image_url": "http://pbs.twimg.com/profile_images/1053387941752369152/r4pGBF3K_normal.jpg",
  "profile_image_url_https": "https://pbs.twimg.com/profile_images/1053387941752369152/r4pGBF3K_normal.jpg",
  "profile_banner_url": "https://pbs.twimg.com/profile_banners/8105922/1544741225",
  "profile_link_color": "FF0324",
  "profile_sidebar_border_color": "FFFFFF",
  "profile_sidebar_fill_color": "EFEFEF",
  "profile_text_color": "333333",
  "profile_use_background_image": true,
  "has_extended_profile": false,
  "default_profile": false,
  "default_profile_image": false,
  "following": true,
  "follow_request_sent": false,
  "notifications": false,
  "translator_type": "none"
}