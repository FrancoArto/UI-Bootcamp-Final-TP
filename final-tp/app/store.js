import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga' 
import AppReducer from './reducers/index'; //Import the reducer

const sagaMiddleware = createSagaMiddleware()
 
export default createStore(AppReducer, applyMiddleware(sagaMiddleware));
