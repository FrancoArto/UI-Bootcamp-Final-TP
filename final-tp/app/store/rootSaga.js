import { all } from 'redux-saga/effects'
import {trendsSaga} from './trends/trendsSaga'
import { userSaga } from './users/userSaga';
import { tweetsSaga } from './tweets/tweetsSaga';

export default function* rootSaga() {
  yield all([
    tweetsSaga(),
    trendsSaga(),
    userSaga()
  ])
}
