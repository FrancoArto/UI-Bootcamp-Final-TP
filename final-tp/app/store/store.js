import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga'
import AppReducer from '../reducers/index'; //Import the reducer
import rootSaga from '../sagas/index';
import reactotron from '../reactotronConfig'

const sagaMonitor = reactotron.createSagaMonitor()


const sagaMiddleware = createSagaMiddleware({ sagaMonitor })

export default reactotron.createStore(AppReducer, applyMiddleware(sagaMiddleware));

sagaMiddleware.run(rootSaga)
