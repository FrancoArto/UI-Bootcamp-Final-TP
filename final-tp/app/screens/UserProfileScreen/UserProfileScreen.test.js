import React from 'react';
import {mount, shallow} from 'enzyme'
import store from '../../store/store';
import ReduxUserProfileScreen,{UserProfileScreen} from './UserProfileScreen';
import userMock from '../../store/users/userMock';
import { tweetsArray } from '../../store/tweets/tweetsMock';



it('renders without crashing', () => {
  const rendered = shallow(<UserProfileScreen userData={userMock} />)
  expect(rendered).toMatchSnapshot();
});

it('renders connected to redux', () => {
  const rendered = shallow(<ReduxUserProfileScreen userData={userMock} store={store} />)
  expect(rendered).toMatchSnapshot();
});

it('renders activity indicator', () => {
  const rendered = shallow(<UserProfileScreen userData={userMock} loadingInfo={true} />)
  expect(rendered).toMatchSnapshot();
});

it('handles tweet press', () => {
  const navigation = {
    navigate: () => {}
  }
  const onTweetPress = jest.fn()
  const wrapper = mount(<UserProfileScreen 
    data={tweetsArray}
    userData={userMock}
    fetchSingleTweetBegin={onTweetPress}
    navigation={navigation}/>)

  const tweet = wrapper.find('Tweet').first();
  tweet.props().onTweetPress();
  expect(onTweetPress).toHaveBeenCalled()
});