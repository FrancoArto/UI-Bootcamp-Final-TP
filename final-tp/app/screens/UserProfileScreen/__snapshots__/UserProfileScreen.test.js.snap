// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders activity indicator 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="small"
  />
</Component>
`;

exports[`renders connected to redux 1`] = `
<UserProfileScreen
  data={Array []}
  error={null}
  fetchSingleTweetBegin={[Function]}
  loadingInfo={false}
  loadingTimeline={false}
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  userData={Object {}}
/>
`;

exports[`renders without crashing 1`] = `
<ScrollView
  style={
    Object {
      "backgroundColor": "#F5F5F5",
      "flex": 1,
    }
  }
>
  <Component
    style={
      Object {
        "flex": 65,
      }
    }
  >
    <UserInfo
      user={
        Object {
          "contributors_enabled": false,
          "created_at": "Fri Aug 10 15:14:11 +0000 2007",
          "default_profile": false,
          "default_profile_image": false,
          "description": "Noticias de Argentina y el mundo las 24 horas. Información de último momento, opinión, análisis e informes especiales.",
          "entities": Object {
            "description": Object {
              "urls": Array [],
            },
            "url": Object {
              "urls": Array [
                Object {
                  "display_url": "clarin.com",
                  "expanded_url": "http://www.clarin.com",
                  "indices": Array [
                    0,
                    22,
                  ],
                  "url": "http://t.co/1RuczZo0FK",
                },
              ],
            },
          },
          "favourites_count": 391,
          "follow_request_sent": false,
          "followers_count": 2810171,
          "following": true,
          "friends_count": 265,
          "geo_enabled": true,
          "has_extended_profile": false,
          "id": 8105922,
          "id_str": "8105922",
          "is_translation_enabled": true,
          "is_translator": false,
          "lang": "es",
          "listed_count": 10833,
          "location": "Buenos Aires, Argentina",
          "name": "Clarín",
          "notifications": false,
          "profile_background_color": "131516",
          "profile_background_image_url": "http://abs.twimg.com/images/themes/theme14/bg.gif",
          "profile_background_image_url_https": "https://abs.twimg.com/images/themes/theme14/bg.gif",
          "profile_background_tile": true,
          "profile_banner_url": "https://pbs.twimg.com/profile_banners/8105922/1544741225",
          "profile_image_url": "http://pbs.twimg.com/profile_images/1053387941752369152/r4pGBF3K_normal.jpg",
          "profile_image_url_https": "https://pbs.twimg.com/profile_images/1053387941752369152/r4pGBF3K_normal.jpg",
          "profile_link_color": "FF0324",
          "profile_location": null,
          "profile_sidebar_border_color": "FFFFFF",
          "profile_sidebar_fill_color": "EFEFEF",
          "profile_text_color": "333333",
          "profile_use_background_image": true,
          "protected": false,
          "screen_name": "clarincom",
          "status": Object {
            "contributors": null,
            "coordinates": null,
            "created_at": "Thu Dec 20 17:06:00 +0000 2018",
            "entities": Object {
              "hashtags": Array [],
              "media": Array [
                Object {
                  "display_url": "pic.twitter.com/pliMCVpKd6",
                  "expanded_url": "https://twitter.com/clarincom/status/1075799489686974467/photo/1",
                  "id": 1075799486948098000,
                  "id_str": "1075799486948098054",
                  "indices": Array [
                    102,
                    125,
                  ],
                  "media_url": "http://pbs.twimg.com/media/Du4B7v0WwAYZ4qc.jpg",
                  "media_url_https": "https://pbs.twimg.com/media/Du4B7v0WwAYZ4qc.jpg",
                  "sizes": Object {
                    "large": Object {
                      "h": 630,
                      "resize": "fit",
                      "w": 1200,
                    },
                    "medium": Object {
                      "h": 630,
                      "resize": "fit",
                      "w": 1200,
                    },
                    "small": Object {
                      "h": 357,
                      "resize": "fit",
                      "w": 680,
                    },
                    "thumb": Object {
                      "h": 150,
                      "resize": "crop",
                      "w": 150,
                    },
                  },
                  "type": "photo",
                  "url": "https://t.co/pliMCVpKd6",
                },
              ],
              "symbols": Array [],
              "urls": Array [
                Object {
                  "display_url": "clarin.com/policiales/ata…",
                  "expanded_url": "https://www.clarin.com/policiales/atacaron-balazos-concejo-deliberante-rosario-mafia-jode_0_t9aX8sX08.html?utm_term=Autofeed&utm_medium=Social&utm_source=Twitter#Echobox=1545324824",
                  "indices": Array [
                    78,
                    101,
                  ],
                  "url": "https://t.co/B8NZ861uoF",
                },
              ],
              "user_mentions": Array [],
            },
            "extended_entities": Object {
              "media": Array [
                Object {
                  "display_url": "pic.twitter.com/pliMCVpKd6",
                  "expanded_url": "https://twitter.com/clarincom/status/1075799489686974467/photo/1",
                  "id": 1075799486948098000,
                  "id_str": "1075799486948098054",
                  "indices": Array [
                    102,
                    125,
                  ],
                  "media_url": "http://pbs.twimg.com/media/Du4B7v0WwAYZ4qc.jpg",
                  "media_url_https": "https://pbs.twimg.com/media/Du4B7v0WwAYZ4qc.jpg",
                  "sizes": Object {
                    "large": Object {
                      "h": 630,
                      "resize": "fit",
                      "w": 1200,
                    },
                    "medium": Object {
                      "h": 630,
                      "resize": "fit",
                      "w": 1200,
                    },
                    "small": Object {
                      "h": 357,
                      "resize": "fit",
                      "w": 680,
                    },
                    "thumb": Object {
                      "h": 150,
                      "resize": "crop",
                      "w": 150,
                    },
                  },
                  "type": "photo",
                  "url": "https://t.co/pliMCVpKd6",
                },
              ],
            },
            "favorite_count": 4,
            "favorited": false,
            "geo": null,
            "id": 1075799489686974500,
            "id_str": "1075799489686974467",
            "in_reply_to_screen_name": null,
            "in_reply_to_status_id": null,
            "in_reply_to_status_id_str": null,
            "in_reply_to_user_id": null,
            "in_reply_to_user_id_str": null,
            "is_quote_status": false,
            "lang": "es",
            "place": null,
            "possibly_sensitive": false,
            "retweet_count": 1,
            "retweeted": false,
            "source": "<a href=\\"https://www.echobox.com\\" rel=\\"nofollow\\">Echobox Social</a>",
            "text": "Atacaron a balazos el Concejo Municipal de Rosario: \\"Con la mafia no se jode\\" https://t.co/B8NZ861uoF https://t.co/pliMCVpKd6",
            "truncated": false,
          },
          "statuses_count": 301743,
          "time_zone": null,
          "translator_type": "none",
          "url": "http://t.co/1RuczZo0FK",
          "utc_offset": null,
          "verified": true,
        }
      }
    />
  </Component>
  <Component
    style={
      Object {
        "flex": 35,
      }
    }
  >
    <FlatList
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      renderItem={[Function]}
      scrollEventThrottle={50}
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </Component>
</ScrollView>
`;
